### ISSUE TITLE
TransitionController - Initial RootViewController view lifecycle not occuring

### ISSUE THREAD

**Issue by markst** [CosmicMind/Material#940]
The initial root view controller view lifecycle functions are not being called on initial load of root view controller. All subsequent lifecycle functions are working as expect.

**Comment by markst** [comment]
This is due to `shouldAutomaticallyForwardAppearanceMethods` being implemented & not calling the appropriate appearance transition functions on initial setup of root view controller:

https://github.com/CosmicMind/Material/pull/938#issuecomment-339204901

**Comment by markst** [comment]
There need to be a feature where by lifecycle functions are performed when updating root view controller directly.

```
self.rootViewController = UIViewController() // View lifecycle not called
self.transition(to: UIViewController(), completion: nil) // View lifecycle called
```

**Comment by danieldahan** [comment]
I see what you are describing. I have an idea of how to make it work correctly. I will give it a try after another task I am working on.

**Comment by markst** [comment]
Not a huge fan of it, but passing the transition logic on as follows, resolved it for me:

```swift

open class TransitionController: UIViewController {

    open override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        self.rootViewController.beginAppearanceTransition(true, animated: animated)
    }
    
    open override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        self.rootViewController.endAppearanceTransition()
    }
    
    open override func viewWillDisappear(_ animated: Bool) {
        self.viewWillDisappear(animated)
        self.rootViewController.beginAppearanceTransition(false, animated: animated)
    }
    
    open override func viewDidDisappear(_ animated: Bool) {
        super.viewDidDisappear(animated)
        self.rootViewController.endAppearanceTransition()
    }
```

**Comment by danieldahan** [comment]
Well I was thinking that in the `shouldAutomaticallyForwardAppearanceMethods` call, we can return true on the first instance.

**Comment by markst** [comment]
yeah sounds fine. trusting it's toggled at the right occasion

**Comment by danieldahan** [comment]
I actually like your solution better as it is clear as to what is happening. With a hidden flag, which I am not a fan of, does not demonstrate clarity when looking through the code. I am going to see how your solution works.

**Comment by danieldahan** [comment]
This is the corresponding commit that will be in the next release https://github.com/CosmicMind/Material/commit/5e8497dbc1f9bfa670b454f4a64a66e8f7ee8844, thank you!

**Comment by danieldahan** [comment]
Please find this fix in [Material 2.12.8](https://github.com/CosmicMind/Material/releases/tag/2.12.8) :)
